### PR TITLE
Bootstrap script fixes and extensions

### PR DESCRIPTION
**Pull request by pmather**
The "sudo apt-get upgrade && sudo apt-get update" commands are in the
wrong order.  The package lists should be updated (via "apt-get
update") before the packages are upgraded (via "apt-get upgrade").  As
things stand, the "sudo apt-get upgrade" will upgrade zero packages
because they will all be up to date already (because the package lists
reference when the VM image was created).

This fix simply updates the package lists and doesn't upgrade the rest
of the packages.  This is so that the installation of Git will succeed. Instead, we postpone updating of all the packages to when it is done
as part of "utils/client_setup.sh".

We also change ubuntu_client_bootstrap.sh so that the RABBITMQ_URL can
optionally be passed in as a parameter when invoking the script.  This
paves the way for provisioning via, e.g., the Vagrant shell provisioner
and passing in the RABBITMQ_URL value when setting up the machine that
way.

### CODE REVIEW

File: utils/fedora_setup.sh [hunk 1]
```diff
@@ -23,11 +32,10 @@ sudo ln -s $(readlink -f /usr/bin/java | sed "s:bin/java::" | sed "s:/jre/::") /
...
 
-cd
-mkdir fedora-data
-sudo chown tomcat7:tomcat7 fedora-data
-sudo sed -i '0,/JAVA_OPTS=".*"/s//JAVA_OPTS=\"-Djava.awt.headless=true -Xmx512m -XX:MaxPermSize=512m -XX:+UseConcMarkSweepGC -Dfcrepo.home=${HOME}\/fedora-data\"/' /etc/default/tomcat7
-sudo sed -i "s:\${HOME}:${HOME}:" /etc/default/tomcat7
+mkdir -p "$FEDORA_DATA"
+sudo chown tomcat7:tomcat7 "$FEDORA_DATA"
+sudo sed -i '0,/JAVA_OPTS=".*"/s//JAVA_OPTS=\"-Djava.awt.headless=true -Xmx512m -XX:MaxPermSize=512m -XX:+UseConcMarkSweepGC -Dfcrepo.home=${FEDORA_DATA}\"/' /etc/default/tomcat7
+sudo sed -i "s:\${FEDORA_DATA}:${FEDORA_DATA}:" /etc/default/tomcat7
```
Comment (dedocibula) [inline utils/fedora_setup.sh:38]: I'm guessing you tested this change right?

File: user_scripts/vm/ubuntu_client_with_rabbitmq_bootstrap.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ sudo rabbitmqctl delete_user guest
 sudo service rabbitmq-server restart
 rm rabbitmq-signing-key-public.asc
 
-sudo apt-get upgrade && sudo apt-get update && sudo apt-get install -y git
+sudo apt-get update && sudo apt-get install -y git
```
Comment (dedocibula) [inline user_scripts/vm/ubuntu_client_with_rabbitmq_bootstrap.sh:19]: As long as this works on AWS I'm fine with the change

**Comment by dedocibula** [comment]
Could you please follow-up with a pull request to or2016 as well?

File: utils/fedora_setup.sh (hunk 1 above)
Comment (pmather) [inline utils/fedora_setup.sh:38]: It has been tested as working on Chameleon Cloud.

File: user_scripts/vm/ubuntu_client_with_rabbitmq_bootstrap.sh (hunk 2 above)
Comment (pmather) [inline user_scripts/vm/ubuntu_client_with_rabbitmq_bootstrap.sh:19]: Did you see the rationale given in the commit message?

Comment (dedocibula) [inline user_scripts/vm/ubuntu_client_with_rabbitmq_bootstrap.sh:19]: Right. Good change :+1:

**Pull request merged**